### DISCUSSION TITLE
Showing root folder node in breadcrumbs

### DISCUSSION THREAD

**Discussion by craigscottuk** [fuma-nama/fumadocs#595] [Q&A] [answered]
Hi,

I've been playing around with Fumadocs for a few days and I really love it. I want to say thank you to Fuma Nama and all the Contributors who have been working on it and making it mega cool thing.

I have a question regarding the breadcrumbs. Would it be possible to have the option to show the breadcrumb item for the root folder (/docs) so the breadcrumb would look like this:

Docs > Thing 1 > 

Maybe a prop 'includeRoot' could make this possible?

![image](https://github.com/user-attachments/assets/ff45121a-5cff-49c8-9151-04cea27fe30b)

The reason I ask is that I'd like to be able to access the index.mdx in the /docs root using the breadcrumbs. For the project I'm working on, it would be really convenient to access this index file when using the app on a mobile device by clicking the breadcrumb item. It would save me from having to access this page through the navigation again.

I'm pretty new to the world of IT. I hope this isn't a silly or time-wasting question.

I really think this is a fantastic document management system. I'd love to contribute when my skills improve.

Thanks a lot for your time! :)

**Comment by fuma-nama** [comment]
There's no an option to configure this. In the past, I removed the root folder node from breadcrumbs in favour of the new design because it was useless most of the time (people could use sidebar  instead, the root index page isn't opened frequently). I will implement this later

**Reply by craigscottuk** [reply]
Hi :)

Thanks a lot for the quick response and for considering reimplementing the root folder node! I understand it probably doesn't make a lot of sense, and I'm sure it is the way it is for good reason. However, if you do find the time to revisit this, it would be greatly appreciated.

Maybe for the time being, anyone else with the skill and knowledge could show me how to make some adjustments to the code to allow the root folder node to appear as a kind of temporary solution? For the record, I did try to do it myself, but the truth is, it is far too advanced for me, and I'm kinda scared to mess about with anything 😅.

I'm using Fumadocs for my personal website because I'm super impressed by it. The Docs Layout is exactly what I've been looking for. It's mega practical, smart, and it looks and feels great to use. Thanks again Fuma Nama and everyone for the hard work and dedication. I'm very grateful 😄.

Regarding my website, I have a lot of content that I want to display and allow the user to explore, like case studies and documentation for two personal projects of mine. I also want to use Fumadocs for my blog too!

I was thinking about having a source for each project and a source for my blog. I thought it would be cool to do it like this (but maybe I'm over complicating...):

``` js
import { map } from '@/.map';
import { createMDXSource } from 'fumadocs-mdx';
import { loader } from 'fumadocs-core/source';

// Source for Custom Bikes (Project 1)
export const {
  getPage: getCustomBikesPage,
  getPages: getCustomBikesPages,
  pageTree: customBikesPageTree,
} = loader({
  baseUrl: '/projects/custom-bikes-system',
  rootDir: 'projects/custom-bikes',
  source: createMDXSource(map),
});

// Source for Quiz (Project 2)
export const {
  getPage: getQuizPage,
  getPages: getQuizPages,
  pageTree: projectQuizTree,
} = loader({
  baseUrl: '/projects/quiz-react-app',
  rootDir: 'projects/quiz',
  source: createMDXSource(map),
});

// Source for Blog
export const {
  getPage: getBlogPage,
  getPages: getBlogPages,
  pageTree: blogPageTree,
} = loader({
  baseUrl: 'blog',
  rootDir: 'blog',
  source: createMDXSource(map),
});
```

So, having the root folder node would take me back to the 'Project Overview' page for that project or the root/index page of the blog would be useful in this case.

For the time being, I have a source called projects, and inside I have the two projects, which I can navigate back to the indexes of each project using the breadcrumbs. But with the blog, it would be nice to be able to navigate back to the blog index.

```js

// Loader for Blog
export const {
  getPage: getBlogPage,
  getPages: getBlogPages,
  pageTree: blogPageTree,
} = loader({
  baseUrl: 'blog',
  rootDir: 'blog',
  source: createMDXSource(map),
});

// Loader for Projects
export const {
  getPage: getProjectsPage,
  getPages: getProjectsPages,
  pageTree: projectsPageTree,
} = loader({
  baseUrl: '/projects',
  rootDir: 'projects',
  source: createMDXSource(map),
});
```
![image](https://github.com/user-attachments/assets/7b2fdea0-ee1c-404f-b49c-233dad05ea07)

It's no issue. It would just be cool to have that option for my case.

Thanks again for your time and assistance!

Best regards, 
Craig

**Reply by fuma-nama** [reply]
It's implemented on the latest version of Fumadocs, also you can use the Links API: https://fumadocs.vercel.app/docs/ui/blocks/links for navigation links

**Reply by craigscottuk** [reply]
Thanks a lot for working on this! I've also been playing around with the the Links API. It is awesome! 😁

I'm still having a bit of bother with the breadcrumbs... maybe it's just me, though... I've added the 'includeRoot' param to the `<DocsPage />` component in `app/blog/[[...slug]]/page.tsx` file but I don't see breadcrumb when I navigate to `/blog/blog-post-one/two/three`.

![image](https://github.com/user-attachments/assets/f9475af4-df95-4851-9986-9ccbbf53b3cb)


```js
// app/blog/[[...slug]]/page.tsx

const MDX = page.data.exports.default;

console.log(blogPageTree);

return (
  <DocsPage
    breadcrumb={{ enabled: true, includeRoot: true }}
    tableOfContentPopover={{ enabled: false }}
    toc={page.data.exports.toc}
    full={page.data.full}
  >
    <DocsBody>
      <h1>{page.data.title}</h1>
      <MDX />
    </DocsBody>
  </DocsPage>
);
```

Here is my blog page tree:

```js
//console.log(blogPageTree);

{
  name: 'Blog',
  children: [
    { type: 'page', name: 'Blog Post 1', url: '/blog/blog-post-one' },
    { type: 'page', name: 'Blog Post 2', url: '/blog/blog-post-two' },
    { type: 'page', name: 'Blog Post 3', url: '/blog/blog-post-three' }
  ]
}
```

There are no folders in my `/blog`. Maybe that's the issue?

![image](https://github.com/user-attachments/assets/053aa960-0a90-42ec-8843-582575fce8a5)

Here is a look at some more of my code:

```js
// content/blog/meta.json

{
  "title": "Blog",
  "pages": ["blog-post-one", "blog-post-two", "blog-post-three"],
  "defaultOpen": false,
  "root": true
}
```
``` json
  "dependencies": {
    "@radix-ui/react-accordion": "^1.2.0",
    "@radix-ui/react-slot": "^1.1.0",
    "@radix-ui/react-tabs": "^1.1.0",
    "embla-carousel-react": "^8.1.6",
    "fumadocs-core": "^12.5.1",
    "fumadocs-mdx": "^8.2.33",
    "fumadocs-ui": "^12.5.1",
    "next": "^14.2.4",
    "next-video": "^1.1.1",
    "react": "^18.3.1",
    "react-dom": "^18.3.1"
  },
```

Thanks a lot for time on this Fuma Nama!!

**Reply by fuma-nama** [reply]
Hmm you didn't include the index.mdx into page tree (the pages property in meta file)? It is missing on your blog page tree

**Reply by fuma-nama** [reply, marked as answer]
Also I will suggest you to add a link component to your page directly instead of using the breadcrumbs component, like how our blog site is implemented

**Reply by craigscottuk** [reply]
Yeah, that's a good idea. The link on the blog looks nice. Thanks a lot for your help and time one this! Really appreciate it.
